### ISSUE TITLE
Split Consume Item menu into tabs

### ISSUE THREAD

**Issue by Yunatan** [CleverRaven/Cataclysm-DDA#30019]
**Is your feature request related to a problem? Please describe.**  
1. Medications shouldn't clog the menu when player decides what to eat
2. Medications should be as easily accessible as food because they are both frequently consumed in bulk
3. Medications shouldn't clog the use\activate menu (player expects it to be populated by his tools)

**Describe the solution you'd like**  
This feature request proposes a compromise to solve all the aforementioned problems.
Introduce configuration option to split **Consume Item** menu into tabs so the player can decide if they want a 'food/drink/meds' tab or a 'food/drink' tab with a separate 'meds' tab or even separate tabs for all three. Menu should immediately repopulate on the same tab when player consumes something to easily use consumables multiple times in a row.

**Describe alternatives you've considered**  
1. Separate menus (too many key pressings)
2. Separate panels (Consume Item menu is already split into columns to differentiate consumables location) 
3. Build tabs onto the existing interface (too painful) 
4. Default-unbound separate menus for players who really need it ([probably gonna go with this for now](https://github.com/CleverRaven/Cataclysm-DDA/pull/30286)) (if this feature gets implemented, the bindings can be redirected to bring up the appropriate tab first)

**Quick paint mock-ups**
Just to simplistically illustrate the concept.

![image](https://user-images.githubusercontent.com/6611291/56864843-f620c400-69cf-11e9-8497-0351b3349eed.png)

![image](https://user-images.githubusercontent.com/6611291/56864847-fe78ff00-69cf-11e9-8410-60bc8742a270.png)

**Additional context**  
Concept created in collaboration with @esotericist.

**Comment by nexusmrsep** [comment]
Would it be also valid to split drinks from food in a dedicated tab in the process?

**Comment by JeanLucVanDamme** [comment]
tl;dr: The UX of the Eat menu would be vastly better with the OP's proposed change plus splitting up food from drink in the same tab. So, if you're very thirsty - press E, tab down to the Drinks category, press enter on the very first item (because, by default, it's got the highest Quench value, exactly what you were looking for.) No more huge overlong lists, less visual clutter, more organization.

I love the OP's suggestion. Splitting drinks from food also seems like a good idea, if it could be accomplished reliably and accurately enough. When your character is thirsty, the point of the Eat menu is to find and select a drink. When he's hungry, you want food. 

But because food, drinks, and meds are currently all bundled together, you must parse a potentially big and visually cluttered list, ignore the irrelevant items, and find (e.g., when thirsty) an item with a high-enough quench value. Compared to the OP's simple mockup, the current process seems unnecessarily tedious.

You'd accomplish the same in-game result with fewer keystrokes, less effort, and in less time, if the Eat menu by default split up food and drinks and sorted each list by its most-relevant attribute (for food, that's Calories; for drinks, that's Quench). 

Of course, it's also true that for some players / some situations, it might be different attribute(s) or even derivations/ratios of the given ones that are most relevant. For example sometimes I don't want the highest-calorie food, but the highest-calorie food without a large negative value for quench. That's the rule I use when my character is hungry but doesn't have any clean water, and I'm just trying to live long enough to get some.

But I think even in these non-default cases, at worst, the proposed change makes no difference in the difficulty of the calculation; and in some cases, it makes the calculation easier. The latter would be true if one zeroed-in on a few of the highest calorie foods to calculate their calorie/quench ratios first, on the logic that very high Calorie values are rarely outweighed by correspondingly high Quench values. 

P.S. You could always add sorting and filtering by other attributes later to further accommodate less-common decision rules. When you start to think about the heuristics and algorithms you use for Eating decisions, it's clear there are several other relevant attributes. Maybe you want clean water, if available, to always top the Drinks list. Maybe you want to filter out unclean water, which you would never drink intentionally anyway. I'm sure everyone can chime in with more examples I haven't thought of.

**Comment by JeanLucVanDamme** [comment]
It could also be fun or rewarding if useful sorting/filtering options are available based on the progression of character skills or traits.

For example, maybe with a high-enough cooking skill, examining a food also shows you which vitamin(s) and how much of each it has. And you'd be able to sort, filter, etc., in the Eat menu based on those attributes. 

By the way, another nice advantage of your guys' ideas is that the ease-of-use changes could spread into other related systems, and reduce tedium for other gameplay tasks. If there's no design downside for, say, allowing a toggle to split up "junk" vs "healthy" food, or food with negative vs positive quench values, you could consider putting these same options into the main inventory screen. 

Now, suddenly, you've eliminated irrelevant mechanical/cognitive tedium that stood in the way of gameplay decisions/actions the user was making already (or trying to make, or would make if they had thought of it.) So, you make this happen easier: "Oh crap, there's a Shocker Brute coming. And I'm too encumbered. I'm gonna die." ... "How about I drop all my junk food... but what about the clean water? No, gotta keep that, it's too risky with High Thirst in the middle of a city with all-empty toilets." You only need a few keystrokes to do this if "junk food" is its own group that can be tabbed-to in the multidrop screen.

Fewer UX hurdles = a greater variety of meaningful choices. Hope the main devs like some stuff in this thread, seems like the potential customization and organization offers only upsides for gameplay.
